docs/Shell-Completion.md

```diff
@@ -12,13 +12,19 @@ To make Homebrew's completions available in `bash`, you must source the definiti
 
 ```sh
 if type brew 2&>/dev/null; then
-  source "$(brew --prefix)/etc/bash_completion.d/*"
-else
-  echo "run: brew install git bash-completion"
+  if [ -d "$(brew --prefix)/etc/bash_completion.d" ];
+  then
+    source "$(brew --prefix)/etc/bash_completion.d/*"
+  fi
+  if [ -f "$(brew --prefix)/etc/profile.d/bash_completion.sh" ];
+  then
+    source "$(brew --prefix)/etc/profile.d/bash_completion.sh"
+  fi
```

Comment:
`[` (and `]`) are archaic. In Bash one should use `[[` and `]]` that, besides other things, allow omitting double quotes. The first `if` would then look like so:

  ```sh
  if [[ -d $(brew --prefix)/etc/bash_completion.d ]]
  then
    source "$(brew --prefix)/etc/bash_completion.d/*"
  fi
```
There two problems with the above code: 1. if there are no files (for whatever reason), the above `*` would become a literal `*` and Bash will fail to find that file and error out. 2. If, on the contrary, the number of files in `bash_completion.d` directory is large, the command might become too long for Bash to handle. So, one has to do something like this:
```sh
for FILE in "$(brew --prefix)/etc/bash_completion.d/*"
do
    [[ -f $FILE ]] || continue
    . "$FILE" || true
done
unset FILE
```
`[[ -f $FILE ]] || continue` to skip `*` "file" when there are no files in that directory.
`|| true` to keep going on error because we don't want to prevent interactive shell from starting even if there is a problem.

For `bash_completion.sh` it might be easier to just source it like so:
```sh
. "$(brew --prefix)/etc/profile.d/bash_completion.sh" 2>/dev/null || true
```

Reply:
Great, thanks @maxim-belkin!